s3upload.js

```diff
@@ -67,126 +73,172 @@ S3Upload.prototype.createCORSRequest = function(method, url, opts) {
             try {
                 result = JSON.parse(xhr.responseText);
-                this.onSignedUrl( result );
+                this.onSignedUrl(result);
             } catch (error) {
                 this.onError('Invalid response from server', file);
                 return false;
             }
             return callback(result);
-        } else if (xhr.readyState === 4 && this.successResponses.indexOf(xhr.status) < 0) {
-            return this.onError('Could not contact request signing server. Status = ' + xhr.status, file);
+        } else if (
+            xhr.readyState === 4 &&
+            this.successResponses.indexOf(xhr.status) < 0
+        ) {
+            return this.onError(
+                'Could not contact request signing server. Status = ' +
+                    xhr.status,
+                file
+            );
         }
     }.bind(this);
     return xhr.send();
 };
 
 S3Upload.prototype.uploadToS3 = function(file, signResult) {
-    var xhr = this.createCORSRequest('PUT', signResult.signedUrl);
+    var xhr;
+    var formData = new FormData();
+
+    if (this.usePostForm) {
+        var fields = signResult.fields;
+        var url = signResult.url;
+        xhr = this.createCORSRequest('POST', url);
+        Object.keys(fields)
+            .sort(argName1 => (argName1 === 'key' ? -1 : 0))
+            .forEach(fieldName =>
+                formData.append(fieldName, fields[fieldName])
+            );
+        formData.append('acl', this.acl);
+        formData.append('Content-type', file.type);
+    } else {
+        xhr = this.createCORSRequest('PUT', signResult.signedUrl);
+        xhr.setRequestHeader('Content-Type', file.type);
+        if (this.contentDisposition) {
+            let disposition = this.contentDisposition;
+            if (disposition === 'auto') {
+                if (file.type.substr(0, 6) === 'image/') {
+                    disposition = 'inline';
+                } else {
+                    disposition = 'attachment';
+                }
+            }
+            const fileName = this.scrubFilename(file.name);
+            xhr.setRequestHeader(
+                'Content-Disposition',
+                `${disposition}; filename="${fileName}"`
+            );
+        }
+        if (signResult.headers) {
+            const signResultHeaders = signResult.headers;
+            Object.keys(signResultHeaders).forEach(key => {
+                const val = signResultHeaders[key];
+                xhr.setRequestHeader(key, val);
+            });
+        }
+        if (this.uploadRequestHeaders) {
+            const uploadRequestHeaders = this.uploadRequestHeaders;
+            Object.keys(uploadRequestHeaders).forEach(key => {
+                const val = uploadRequestHeaders[key];
+                xhr.setRequestHeader(key, val);
+            });
+        } else {
+            xhr.setRequestHeader('x-amz-acl', this.acl);
+        }
+    }
+
+    formData.append('file', file);
```

Comment:
out of curiosity: does the put with signedurl also work after this changes?

Reply:
no before inside a specific condition, it's a common code

Reply:
it's working for both situation, createPresignedPost and signedUrl